docs/DeveloperGuide.md

```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
 
     Use case ends.
 
 **Extensions**
-
-* 2a. The list is empty.
-  
-  * Use case ends.
-     
-* 3a. The given index is invalid.
-    
-    * 3a1. Productiv shows an error message.
-    
-      Use case resumes at step 2.
+* 2a. Productiv detects an error in the command.
+    * 2a1. Productiv displays an error message.
+    * 2a2. User enters the command again.
+    * Steps 2a1-2a2 are repeated until the command entered is correct.
       
+      Use case resumes from step 3.
+    
+**Use case: UC15 - View a deliverable**
 
+**Precondition(s):**
+* User is in the deliverable mode.
+* Deliverable to view exists.
 
+**MSS**
+1. User chooses to view a deliverable.
+2. User enters the command to view a deliverable into the command box.
+3. Productiv displays the deliverable in the view box.
 
-#### DELIVERABLES
+    Use case ends.
 
-**Use case: `UC20 - Add Deliverable`**
+**Extensions**
+* 2a. Productiv detects an error in the command.
+    * 2a1. Productiv displays an error message.
+    * 2a2. User enters the command again.
+    * Steps 2a1-2a2 are repeated until the command entered is correct.
+      
+      Use case resumes from step 3.  
+    
+**Use case: UC16 - Find deliverables**
 
 **Precondition(s):**
-* **`User is in the Deliverable mode`** 
+* User is in the deliverable mode.
+* User has keyword(s) for Productiv to find deliverables with.
 
 **MSS**
+1. User chooses to find deliverables.
+2. User enters the command and keyword(s) to find deliverables into the command box.
+3. Productiv returns all matching deliverables in the list box.
 
-1. User chooses to add a deliverable.
-1. User enters the command to add a deliverable into the input box.
-1. Productiv adds the deliverable for tracking.
+    Use case ends.
+
+**Extensions**
+* 2a. Productiv detects an error in the command.
+    * 2a1. Productiv displays an error message.
+    * 2a2. User enters the command again.
+    * Steps 2a1-2a2 are repeated until the command entered is correct.
+      
+      Use case resumes from step 3.  
+      
+**Use case: UC17 - List all deliverables**
+
+**Precondition(s):**
+* User is in the deliverable mode.
+* User has at least one existing deliverable in Productiv.
+
+**MSS**
+1. User chooses to list all deliverables.
+2. User enters the command to list all deliverables into the command box.
+3. Productiv lists out all deliverables in the list box.
 
     Use case ends.
 
 **Extensions**
+* 2a. Productiv detects an error in the command.
+    * 2a1. Productiv displays an error message.
+    * 2a2. User enters the command again.
+    * Steps 2a1-2a2 are repeated until the command entered is correct.
+      
+      Use case resumes from step 3.  
+      
+**Use case: UC18 - Delete a deliverable**
 
-* 2a. Produtiv detects an error in the command.
+**Precondition(s):**
+* User is in the deliverable mode.
+* Deliverable to delete exists.
+
+**Guarantee(s):** 
+* Upon successful command, the OCP will be updated accordingly.
```

Comment:
actually if you're gonna say this, probably should say calendar will also be updated